src/index.tsx

```diff
@@ -528,8 +526,48 @@ export async function validateServerFormData<T extends FormDefinition>(
   >;
   await Promise.all(
     entries.map(async ([inputName, inputDef]) => {
-      let values = formData.getAll(inputName);
-      if (formData.has(inputName)) {
```

Comment:
Updated this logic to parse according to the schema instead of what came in from `formData`.  so only inputs that are defined as having multiple values will use `getAll()`